src/main/js/util/fieldMapping.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import SelectInput from 'field/input/SelectInput';
+import TextInput from 'field/input/TextInput';
+import TextArea from 'field/input/TextArea';
+import PasswordInput from 'field/input/PasswordInput';
+import NumberInput from 'field/input/NumberInput';
+import CheckboxInput from 'field/input/CheckboxInput';
+import ReadOnlyField from 'field/ReadOnlyField';
+import * as FieldModelUtilities from 'util/fieldModelUtilities';
+
+export function getField(fieldType, props) {
+    switch (fieldType) {
+        case 'Select':
```

Comment:
You could look into a static map containing references to functions that you can lookup by field type.
You can look to see if that makes the code more readable than the switch statement.

```
export function buildTextInput(props) {
    return <TextInput {...props} />
}

export static const builders = { 
    'TextInput': buildTextInput,
     ...
}

export function getField(fieldType, props) {
    const builder = builders[fieldType];
    return builder(props);
}
```